packages/build/src/plugins/config_prop_name.js

```diff
@@ -16,27 +17,11 @@ const isArrayItem = function (key) {
 }
 
 // Check if a property name is dynamic, such as `functions.{functionName}`
-const isDynamicObjectProp = function (keys, index) {
-  return (
-    index !== 0 &&
-    DYNAMIC_OBJECT_PROPS.has(serializeKeys(keys.slice(0, index))) &&
-    !NON_DYNAMIC_OBJECT_PROPS.has(serializeKeys(keys.slice(0, index + 1)))
-  )
-}
-
-const serializeKeys = function (keys) {
-  return keys.map(String).join('.')
+const isDynamicObjectProp = function (propName) {
+  return DYNAMIC_OBJECT_PROPS.has(propName)
 }
 
 // Properties with dynamic children
-const DYNAMIC_OBJECT_PROPS = new Set(['functions'])
-// Children properties which should not be considered dynamic
-const NON_DYNAMIC_OBJECT_PROPS = new Set([
-  'functions.directory',
```

Comment:
All `functions.*` are now handled the same way since the `functions.*` logic happens inside `@netlify/config` instead. Thanks to this, this file can be simplified a lot.